### PR TITLE
feat(Partner Management APIs): add new service to project

### PR DESCRIPTION
**Pull request by manukm-ibm**
## PR summary

Adding new partner API to fetch resource usage reports.
TRI: https://github.ibm.com/BSS/architecture/blob/master/design-docs/cosell/partner.md
Issue: https://github.ibm.com/BSS/metering-deliverables/issues/2063

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [x] Yes
- [ ] No

### CODE REVIEW

File: partnermanagementapisv1/partner_management_apis_v1.go [hunk 1]
```diff
@@ -0,0 +1,1771 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * IBM OpenAPI SDK Code Generator Version: 3.87.0-91c7c775-20240320-213027
+ */
+
+// Package partnermanagementapisv1 : Operations and models for the PartnerManagementAPIsV1 service
+package partnermanagementapisv1
```
Comment (padamstx) [inline partnermanagementapisv1/partner_management_apis_v1.go:22]: I think a better package name would be `partnermanagementv1`. The `apis` part is sort of redundant and unnecessary I think.  If you want to change this (probably no better time to do it than now before you actually introduce the new service :joy:), you could follow these steps:

1. Modify the `info.title` field in the API definition to change it to "Partner Management"

OR

Add the `info.x-alternate-name`field to the API definition like this:
```
openapi: 3.0.0
info:
  title: "Partner Management APIs"
  x-alternate-name: "Partner Management"
  version: 1.0.0
```

2. Re-generate your SDK code. The SDK generator will use the x-alternate-name value (if present) when it needs to compute the package name (e.g. `partnermanagementv1`).

**Review by padamstx — CHANGES_REQUESTED** [review]
Please update the README.md (service table) to remove the entries for the two services being removed and add a new entry for the new service that you're adding.
Also, I recommended a change to the package name to shorten it up a bit.

File: partnermanagementapisv1/partner_management_apis_v1.go (hunk 1 above)
Comment (manukm-ibm) [reply partnermanagementapisv1/partner_management_apis_v1.go]: Thank you Phil for the review. Updated the code

**Comment by manukm-ibm** [comment]
Integration test results


Example run results

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**